std/algorithm/comparison.d

```diff
@@ -705,25 +705,24 @@ int cmp(alias pred = "a < b", R1, R2)(R1 r1, R2 r2) if (isSomeString!R1 && isSom
 
 // equal
 /**
-Compares two ranges for equality, as defined by predicate $(D pred)
+Compares two ore more ranges for equality, as defined by predicate $(D pred)
 (which is $(D ==) by default).
 */
 template equal(alias pred = "a == b")
 {
-    enum isEmptyRange(R) =
-        isInputRange!R && __traits(compiles, {static assert(R.empty);});
-
-    enum hasFixedLength(T) = hasLength!T || isNarrowString!T;
+    import std.typecons : allSatisfy;
 
     /++
-    Compares two ranges for equality. The ranges may have
-    different element types, as long as $(D pred(r1.front, r2.front))
-    evaluates to $(D bool).
-    Performs $(BIGOH min(r1.length, r2.length)) evaluations of $(D pred).
+
+    This function compares the ranges $(D r) and $(D ss) for equality. The
```

Comment:
"ss" is surprising, we use "rs" for "plural of r"

Reply:
Done.